### PR TITLE
Fix dereferencing printable types

### PR DESCRIPTION
**Pull request by sarge**
Fixes I hope the case where a non printable type - struct is being returned.

Apologies for the build on the modules branch, but that is how I do all development now.

### CODE REVIEW

**Comment by annismckenzie** [comment]
I merged #121 just now – can you rebase this, please? :)

File: eval.go [hunk 1]
```diff
@@ -1091,14 +1091,32 @@ func (st *Runtime) evalBaseExpressionGroup(node Node) reflect.Value {
...
+			case reflect.String,
+				reflect.Bool,
+				reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64,
+				reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64,
+				reflect.Float32, reflect.Float64:
+				return rTest
+			}
+
+			if dereferenceLimit == 0 {
+				return reflect.ValueOf("<nil>")
```
Comment (sauerbraten) [inline eval.go]: doesn't this make it impossible to ever do anything at all with, say, a `***MyStruct`?

I mean, hopefully never needs to have three levels of indirection, but we shouldn't make it impossible to work with those. I'd say we return the original value in this case. This function is not only used to render values into the output string, but also to evaluate the base expression in chain expressions, which is usually an identifier, meaning this code will run when evaluating `{{a.Name()}}` where `a` might be a pointer to a variable of `type A **Foo` or something like that.

**Comment by sauerbraten** [comment]
I thought about it a bit more, and I actually believe that template evaluation is the wrong place for implicit pointer dereferentiation. Instead, this should be done when printing the result of an evaluation, i.e. in this case in https://github.com/CloudyKit/fastprinter. I will prepare a pull request there.

**Comment by tooolbox** [comment]
> I thought about it a bit more, and I actually believe that template evaluation is the wrong place for implicit pointer dereferentiation. Instead, this should be done when printing the result of an evaluation, i.e. in this case in https://github.com/CloudyKit/fastprinter.

You know, that had occurred to me actually.  So I guess this means that if we merge https://github.com/CloudyKit/fastprinter/pull/2 then this PR should remove the indirection entirely, no?

**Comment by sarge** [comment]
Great idea, makes much better sense to me too. I was a bit iffy about my approach to begin with.

**Pull request closed without merging**